helix-core/src/main/java/org/apache/helix/controller/rebalancer/AutoRebalancer.java

```diff
@@ -64,7 +64,12 @@ public IdealState computeNewIdealState(String resourceName,
 
     LOG.info("Computing IdealState for " + resourceName);
 
-    List<String> partitions = new ArrayList<>(currentIdealState.getPartitionSet());
+    // This is part of the backward compatible workaround to fix
+    // https://github.com/apache/helix/issues/940.
+    // TODO: remove the workaround once we are able to apply the simple fix without majorly
```

Comment:
When do you think we can fix it permanently?

Reply:
There are 2 plans in my mind:
1. Encourage customers to use WAGED. Needs to wait until WAGED is stable.
2. Adding a CRUSHedV2 with the sorting logic, and leave V1 as it is now. Then deprecate V1. For this one, I would prefer to do it after we find all issues in the CRUSHed otherwise we might have V3 later.

After compare between these 2 options, I personally prefer the first option. That's why I actually removed the CRUSHedV2 after I added it to my PR. What's your opinion?